src/controller/java/src/matter/onboardingpayload/QRCodeOnboardingPayloadParser.kt

```diff
@@ -60,11 +68,64 @@ class QRCodeOnboardingPayloadParser(private val mBase38Representation: String) {
       throw UnrecognizedQrCodeException("Invalid argument")
     }
 
-    // TODO: populate TLV optional fields
+    populateTLV(outPayload, buf, indexToReadFrom)
 
     return outPayload
   }
 
+  private fun populateTLV(
+    payload: OnboardingPayload,
+    payloadData: ArrayList<Byte>,
+    index: AtomicInteger
+  ) {
+    val bitsLeftToRead = (payloadData.count() * 8) - index.get()
```

Comment:
Used the QRCodeSetupPayloadParser.cpp for reference: https://github.com/project-chip/connectedhomeip/blob/927962863180270091c1694d4b1ce2e9ea16b8b5/src/setup_payload/QRCodeSetupPayloadParser.cpp#L265